src/main/kotlin/org/rust/ide/intentions/UnwrapLambdaExprIntention.kt

```diff
@@ -0,0 +1,47 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Condition
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.RustBlockExprElement
+import org.rust.lang.core.psi.RustElementFactory
+import org.rust.lang.core.psi.RustLambdaExprElement
+import org.rust.lang.core.psi.util.parentOfType
+
+class UnwrapLambdaExprIntention : PsiElementBaseIntentionAction() {
+    override fun getText() = "Remove braces from lambda expression"
+    override fun getFamilyName() = getText()
+    override fun startInWriteAction() = true
+
+    override fun invoke(project: Project, editor: Editor, element: PsiElement) {
+        val lambdaExpr = element.parentOfType<RustLambdaExprElement>() ?: return
+        val block = PsiTreeUtil.findFirstParent(element, blockOrLambda) as? RustBlockExprElement ?: return
+        val paramsStr = lambdaExpr.parameters.text
+        val bodyStr = block.block?.expr?.text ?: return
+        val unbraced = RustElementFactory.createUnbracedLambdaExpr(project, paramsStr, bodyStr) ?: return
+        lambdaExpr.replace(unbraced)
+    }
+
+    override fun isAvailable(project: Project, editor: Editor, element: PsiElement): Boolean {
+        if (!element.isWritable) return false
+
+        // find first parent which is either block or lambda
+        // if we get lambda, than return false as we are inside "unblocked" lambda
+        val block = PsiTreeUtil.findFirstParent(element, blockOrLambda) as? RustBlockExprElement ?: return false
+
+        // check whether our block's parent is lambda to avoid false positives
+        // like nested blocks in lambda (we are not interested in this case as we can
+        // be sure that cursor isn't at brace; well, I don't know what about `|x| {<caret>{ x }}`,
+        // but it's possibly rare edge case), or just some random block in code
+        if (block.parent !is RustLambdaExprElement) return false
+        val body = block.block?.expr ?: return false
+        return block.text.removePrefix("{").removeSuffix("}").trim() == body.text
```

Comment:
cool! that's what i was looking for 😂  can not find a PSI level api like `lbrace` and `rbrace`, thx!